sfge/src/main/java/com/salesforce/rules/usewithsharingondatabaseoperation/UseWithSharingOnDatabaseOperationHandler.java

```diff
@@ -0,0 +1,61 @@
+package com.salesforce.rules.usewithsharingondatabaseoperation;
+
+import com.salesforce.exception.ProgrammingException;
+import com.salesforce.graph.ApexPath;
+import com.salesforce.graph.symbols.CloningSymbolProvider;
+import com.salesforce.graph.symbols.SharingPolicySymbolProviderVertexVisitor;
+import com.salesforce.graph.symbols.SymbolProvider;
+import com.salesforce.graph.vertex.*;
+import com.salesforce.graph.visitor.ApexPathWalker;
+import com.salesforce.rules.Violation;
+import com.salesforce.rules.avoiddatabaseoperationinloop.DatabaseOperationUtil;
+import com.salesforce.rules.ops.boundary.SharingPolicyBoundaryDetector;
+import java.util.Set;
+import org.apache.tinkerpop.gremlin.process.traversal.dsl.graph.GraphTraversalSource;
+
+public class UseWithSharingOnDatabaseOperationHandler {
+
+    public boolean test(BaseSFVertex vertex) {
+        return DatabaseOperationUtil.isDatabaseOperation(vertex);
+    }
+
+    public Set<Violation.PathBasedRuleViolation> detectViolations(
+            GraphTraversalSource g, ApexPath path, BaseSFVertex dbOpVertex) {
+
+        final BaseSFVertex sourceVertex = path.getMethodVertex().orElse(null);
+
+        if (!(DatabaseOperationUtil.isDatabaseOperation(dbOpVertex))) {
```

Comment:
Is this necessary?

Reply:
This came from the MMSLR "scaffolding." Isn't it good, since we're expecting a database operation, but can't make the parameter strongly typed to such?